docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,1155 @@
+
+:::info
+`expect` is one the functions that's only available in the REPL but there are many more, check them out <a href="https://pact-language.readthedocs.io/en/latest/pact-functions.html#repl-only-functions">here</a>.
+:::
+
+And more tests:
+
+```clojure
+(begin-tx)
+(use election)
+;; test if votes count for candidate "1" is initialized with 0
+(expect "votes for Candidate A initialized" (get-votes "1") 0)
+;; test if votes count for candidate "2" is initialized with 0
+(expect "votes for Candidate B initialized" (get-votes "2") 0)
+(commit-tx)
+
+(begin-tx)
+(use election)
+(env-sigs [{ "key": "alice-key", "caps": [(coin.GAS), (election.ACCOUNT-OWNER "alice")]}])
+
+;; test if votes count for candidate "1" is correctly increased by 1
+(let ((count (get-votes "1")))
+  (vote "alice" "1")
+  (expect "votes count is increased by 1" (get-votes "1") (+ count 1)))
+
+(expect "voted event"
+  [ { "name": "election.VOTED", "params": ["1"], "module-hash": (at 'hash (describe-module "election"))}]
+  (env-events true))
+
+(env-sigs [{ "key": "bob-key", "caps": [(coin.GAS), (election.ACCOUNT-OWNER "bob")]}]}])
+;; test if votes count for candidate "2" is correctly increased by 1
+(let ((count (get-votes "2")))
+  (vote "bob" "2")
+  (expect "votes count is increased by 1" (get-votes "2") (+ count 1)))
+
+(expect "voted event"
+  [ { "name": "election.VOTED",
+      "params": ["2"],
+      "module-hash": (at 'hash (describe-module "election"))
+    }
+  ]
+  (env-events true))
+
+(expect-failure "Double voting not allowed" (vote "bob" "1"))
+
+(commit-tx)
+```
+Here we are testing that the `vote` function correctly increments the vote count and the `VOTED` event is emitted. We are also checking if we correctly handled the double voting scenario.
+
+:::tip
+Can you think of some cases that we didn't cover? Hint: ACCOUNT-OWNER
+
+Try to write a test that validates that only the correct owner of an account can vote.
+:::
+
+ Notice the `let` construct which is helpful when you need to bind some variables to be in the same scope as some other logic that uses them. In our case we first loaded the number of votes and binded the result to `count` variable which we compared with the new count after submitting a vote. You can read more about `let` and `let*` <a href="https://pact-language.readthedocs.io/en/latest/pact-reference.html#let"> here</a>.
+
+Run those tests again to make sure everything works as expected:
+
+```clojure
+$ pact
+pact> (load "election.repl")
+```
+
+:::info
+The REPL preserves state between subsequent runs unless the optional parameter `reset` is set to true `(load "election.repl" true)`.
+:::
+
+### Gas Station
+
+A unique feature of Kadena is the ability to allow gas to be paid by a different entity than the one who initiated the transaction. This entity is what we call a *gas station*.
+
+ In our voting app this will allow users to submit votes without paying for gas, instead gas will be subsidized by the gas station.
```

Comment:
Maybe add a sentence: "In short, this means that miners will still be paid, but our users can vote for free."